src/jit/codegenarm64.cpp

```diff
@@ -2722,114 +3022,104 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
-                }
-                genProduceReg(treeNode);
-            }
+            genUpdateLife(varNode);
+
+            varDsc->lvRegNum = REG_STK; 
         }
         break;
 
-    case GT_RETFILT:
-        // A void GT_RETFILT is the end of a finally. For non-void filter returns we need to load the result in
-        // the return register, if it's not already there. The processing is the same as GT_RETURN.
-        if (targetType != TYP_VOID)
+    case GT_STORE_LCL_VAR:
         {
-            // For filters, the IL spec says the result is type int32. Further, the only specified legal values
-            // are 0 or 1, with the use of other values "undefined".
-            assert(targetType == TYP_INT);
-        }
+            GenTreeLclVarCommon* varNode = treeNode->AsLclVarCommon();
 
-        __fallthrough;
+            unsigned   varNum = varNode->gtLclNum;         assert(varNum < compiler->lvaCount);
+            LclVarDsc* varDsc = &(compiler->lvaTable[varNum]);
+            unsigned   offset = 0;
 
-    case GT_RETURN:
-        {
-            GenTreePtr op1 = treeNode->gtOp.gtOp1;
-            if (targetType == TYP_VOID)
+            // Ensure that lclVar nodes are typed correctly.
+            assert(!varDsc->lvNormalizeOnStore() || targetType == genActualType(varDsc->TypeGet()));
+
+            GenTreePtr data = treeNode->gtOp.gtOp1->gtEffectiveVal();
+
+            // var = call, where call returns a multi-reg return value
+            // case is handled separately.
+            if (data->gtSkipReloadOrCopy()->IsMultiRegCall())
             {
-                assert(op1 == nullptr);
+                genMultiRegCallStoreToLocal(treeNode);
             }
             else
             {
-                assert(op1 != nullptr);
-                noway_assert(op1->gtRegNum != REG_NA);
-
-                genConsumeReg(op1);
-
-                regNumber retReg = varTypeIsFloating(treeNode) ? REG_FLOATRET : REG_INTRET;
+                genConsumeRegs(data);
 
-                bool movRequired = (op1->gtRegNum != retReg);
+                regNumber dataReg = REG_NA;
+                if (data->isContainedIntOrIImmed())
+                {
+                    assert(data->IsIntegralConst(0));
+                    dataReg = REG_ZR;
+                }
+                else
+                {
+                    assert(!data->isContained());
+                    genConsumeReg(data);
+                    dataReg = data->gtRegNum;
+                }
+                assert(dataReg != REG_NA);
 
-                if (!movRequired)
+                if (targetReg == REG_NA)  // store into stack based LclVar
                 {
-                    if (op1->OperGet() == GT_LCL_VAR)
+                    // Only true gtLclVar subclass nodes currently have a gtLclILoffs instance field
+                    // 
+                    if (treeNode->gtOper != GT_STORE_LCL_FLD)
```

Comment:
I think that this is no longer needed, since you split out the GT_STORE_LCL_FLD case?